lib/rubocop/cop/lint/literal_as_condition.rb

```diff
@@ -248,7 +248,7 @@ def correct_if_node(node, cond)
             add_offense(cond) do |corrector|
               corrector.replace(node, "else\n  #{node.else_branch.source}")
             end
-          elsif result
+          elsif result && node.if_branch
```

Comment:
That's just my two cents. Can you align the order of conditions with those in L243 and L247?

```suggestion
          elsif node.if_branch && result
```

Reply:
Done.